core_lib/data_layers/data_access/db/crud/crud_soft_delete_token_data_access.py

```diff
@@ -0,0 +1,36 @@
+from datetime import datetime
+
+from core_lib.data_layers.data_access.data_access import DataAccess
+from core_lib.data_layers.data.handler.sql_alchemy_data_handler_registry import SqlAlchemyDataHandlerRegistry
+from core_lib.data_layers.data_access.db.crud.crud import CRUD
+from core_lib.error_handling.not_found_decorator import NotFoundErrorHandler
+from core_lib.rule_validator.rule_validator import RuleValidator
+
+
+class CRUDSoftDeleteWithTokenDataAccess(DataAccess, CRUD):
+    def __init__(self, db_entity, db: SqlAlchemyDataHandlerRegistry, rule_validator: RuleValidator):

```

Comment:
make `rule_validator` optional here 
+ also make it optional in other CRUD data access's

Reply:
optional `rule_validator: RuleValidator = None`